tests/Functional/Driver/Mysqli/ResultTest.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Tests\Functional\Driver\Mysqli;
+
+use Doctrine\DBAL\Driver\Mysqli\Result;
+use Doctrine\DBAL\Schema\Table;
+use Doctrine\DBAL\Tests\FunctionalTestCase;
+use Doctrine\DBAL\Tests\TestUtil;
+use mysqli;
+use mysqli_driver;
+use mysqli_sql_exception;
+
+use function assert;
+use function sprintf;
+
+use const MYSQLI_REPORT_ERROR;
+use const MYSQLI_REPORT_OFF;
+use const MYSQLI_REPORT_STRICT;
+
+/** @requires extension mysqli */
+final class ResultTest extends FunctionalTestCase
+{
+    private const TABLE_NAME = 'result_test_table';
+
+    private mysqli $nativeConnection;
+
+    protected function setUp(): void
+    {
+        if (! TestUtil::isDriverOneOf('mysqli')) {
+            self::markTestSkipped('This test requires the mysqli driver.');
+
+            return;
+        }
+
+        $nativeConnection = $this->connection->getNativeConnection();
+
+        assert($nativeConnection instanceof mysqli);
+
+        $this->nativeConnection = $nativeConnection;
+
+        $table = new Table(self::TABLE_NAME);
+        $table->addColumn('my_col_1', 'integer', ['notnull' => true]);
+
+        $this->dropAndCreateTable($table);
+    }
+
+    protected function tearDown(): void
+    {
+        $this->dropTableIfExists(self::TABLE_NAME);
+    }
+
+    public function testSuccessfulRowCountFromAffectedRows(): void
+    {
+        $mysqliStmt = $this->nativeConnection
+            ->prepare(sprintf('INSERT INTO %s VALUES (7);', self::TABLE_NAME));
+        $mysqliStmt->execute();
+
+        self::assertSame(1, $mysqliStmt->affected_rows);
+        self::assertSame(1, (new Result($mysqliStmt))->rowCount());
+    }
```

Comment:
This test feels like it tests something that's already covered elsewhere.

Reply:
I think you're right:
https://github.com/doctrine/dbal/blob/4e673be0ecad4ee4fd68c55f686edc098df6a047/tests/Functional/WriteTest.php#L61-L70